### PR TITLE
LANG-1402: Null/index safe get methods for ArrayUtils

### PR DESCRIPTION
**Pull request by MarkDacek**
@chtompki 
If this is deemed worthy or needs improvement - let me know.

### CODE REVIEW

**Comment by sebbASF** [comment]
User code has to cast the return to make use of it; that is not ideal.

**Comment by MarkDacek** [comment]
I added your suggested method and refactored the other to use generics.

File: src/main/java/org/apache/commons/lang3/ArrayUtils.java [hunk 1]
```diff
@@ -8672,4 +8672,53 @@ public static void shuffle(final double[] array, final Random random) {
...
+     * @return The element in the array at the specified index, or the given argument if it is ill-formatted
+     * @since 3.8
+     */
+    public static <T> T get(T[] array, int index, T defaultReturn){
+        if(getLength(array) == 0 || array.length <= index){
+            return defaultReturn;
+        }
+
+        if(index < 0 ){
+            index = 0;
```
Comment (sebbASF) [inline src/main/java/org/apache/commons/lang3/ArrayUtils.java]: Surely a negative index should return the default?

File: src/main/java/org/apache/commons/lang3/ArrayUtils.java [hunk 2]
```diff
@@ -8672,4 +8672,53 @@ public static void shuffle(final double[] array, final Random random) {
...
+        }
+
+        return array[index];
+    }
+
+    /**
+     * Gets an element from the array if the array is non-null and appropriately long, otherwise returns the specified value
```
Comment (sebbASF) [inline src/main/java/org/apache/commons/lang3/ArrayUtils.java]: Does not agree with code

File: src/main/java/org/apache/commons/lang3/ArrayUtils.java [hunk 3]
```diff
@@ -8672,4 +8672,53 @@ public static void shuffle(final double[] array, final Random random) {
...
+     * @param <T> the component type of the array
+     * @param array   the array holding the desired element
```
Comment (sebbASF) [inline src/main/java/org/apache/commons/lang3/ArrayUtils.java]: May be null

File: src/main/java/org/apache/commons/lang3/ArrayUtils.java [hunk 4]
```diff
@@ -8672,4 +8672,53 @@ public static void shuffle(final double[] array, final Random random) {
             swap(array, i - 1, random.nextInt(i), 1);
         }
     }
+
+    /**
+     * Gets an element from the array if the array is non-null and appropriately long, otherwise returns null
+     * @param <T> the component type of the array
+     * @param array   the array holding the desired element
+     * @param index  the index of the element in the array
+     * @return The element in the array at the index, or null if it is ill-formatted
```
Comment (sebbASF) [inline src/main/java/org/apache/commons/lang3/ArrayUtils.java]: What does ill-formatted mean?
Can also return null if the array element is present and null.

I don't think this method is sufficiently useful to be worth adding

File: src/main/java/org/apache/commons/lang3/ArrayUtils.java [hunk 5]
```diff
@@ -8672,4 +8672,53 @@ public static void shuffle(final double[] array, final Random random) {
...
+        return get(array, index, null);
+    }
+
+    /**
+     * Gets an element from the array if the array is non-null and appropriately long, otherwise returns the specified value
+     * @param <T> the component type of the array
+     * @param array   the array holding the desired element
+     * @param index  the index of the element in the array
+     * @param defaultReturn the object to be returned if the array is null or shorter than the index
+     * @return The element in the array at the specified index, or the given argument if it is ill-formatted
```
Comment (sebbASF) [inline src/main/java/org/apache/commons/lang3/ArrayUtils.java]: ill-formatted?

Again, I don't see the point of this method.
It's a bit more flexible than the original method.
But it's still only useful for the case where you can choose a default value that does not appear in the array, and you still have to check whether the return value is the default or not.

File: src/main/java/org/apache/commons/lang3/ArrayUtils.java (hunk 2 above)
Comment (MarkDacek) [reply src/main/java/org/apache/commons/lang3/ArrayUtils.java]: Sorry, mistaken copy-paste there. Feel free to use this as a tutorial for how NOT to contribute.

File: src/main/java/org/apache/commons/lang3/ArrayUtils.java (hunk 1 above)
Comment (MarkDacek) [reply src/main/java/org/apache/commons/lang3/ArrayUtils.java]: Fixed.

File: src/main/java/org/apache/commons/lang3/ArrayUtils.java (hunk 5 above)
Comment (MarkDacek) [reply src/main/java/org/apache/commons/lang3/ArrayUtils.java]: My thought process was something similar to this https://github.com/apache/commons-lang/blob/master/src/main/java/org/apache/commons/lang3/StringUtils.java#L7456-L7458

I would understand if Strings are understood well-enough to merit a true default value and generic arrays aren't.

**Comment by MarkDacek** [comment]
@sebbASF Thanks for your patience and efforts to make this a worthy contribution. I have removed the other methods per your suggestion.

**Pull request merged**